src/components/App.tsx

```diff
@@ -2741,6 +2743,86 @@ class App extends React.Component<AppProps, AppState> {
     }
   };
 
+  private handleEraser = (
+    event: PointerEvent,
+    pointerDownState: PointerDownState,
+    scenePointer: { x: number; y: number },
+  ) => {
+    const idsToUpdate: Array<string> = [];
+
+    const distance = distance2d(
+      pointerDownState.lastCoords.x,
+      pointerDownState.lastCoords.y,
+      scenePointer.x,
+      scenePointer.y,
+    );
+    const threshold = 10 / this.state.zoom.value;
+    const point = pointerDownState.lastCoords;
+    let samplingInterval = 0;
+    while (samplingInterval < distance) {
```

Comment:
- you're hitbox-testing the original position twice (once in previous invocation, once at the start of the loop)
- I'd also check if the loop if the interpolation is needed at all. First, I'd go from end to start: drop the `samplingInterval` variable and just use the `distance`, starting from the end and subtracting `threshold` from it on each loop iteration, and making the loop condition to be `distance >= threshold`. This way you don't loop unnecessarily and reduce the variables.

Reply:
hitbox-testing original position twice ? I am hitbox testing the last coords here and scene pointer at the end of the loop as the condition is `sampleInterval < distance`, I can club that into the loop as well using `sampleInterval <= distance`. However we can remove `sampleInterval` and go in reverse direction so that extra variable is not needed